aws/resource_aws_spot_fleet_request.go

```diff
@@ -676,19 +776,97 @@ func buildAwsSpotFleetLaunchSpecifications(
 	return specs, nil
 }
 
+func buildLaunchTemplateConfigs(d *schema.ResourceData) ([]*ec2.LaunchTemplateConfig, error) {
+	launchTemplateConfigs := d.Get("launch_template_configs").(*schema.Set)
+	configs := make([]*ec2.LaunchTemplateConfig, 0)
+
+	for _, launchTemplateConfig := range launchTemplateConfigs.List() {
+
+		ltc := &ec2.LaunchTemplateConfig{}
+
+		ltcMap := launchTemplateConfig.(map[string]interface{})
+
+		//launch template spec
+		if v, ok := ltcMap["launch_template_specification"]; ok {
+			vL := v.([]interface{})
+			lts := vL[0].(map[string]interface{})
+
+			flts := &ec2.FleetLaunchTemplateSpecification{}
+
+			if v, ok := lts["id"].(string); ok && v != "" {
+				flts.LaunchTemplateId = aws.String(v)
+			}
+
+			if v, ok := lts["name"].(string); ok && v != "" {
+				flts.LaunchTemplateName = aws.String(v)
+			}
+
+			if v, ok := lts["version"].(string); ok && v != "" {
+				flts.Version = aws.String(v)
+			}
+
+			ltc.LaunchTemplateSpecification = flts
+
+		}
+
+		if v, ok := ltcMap["overrides"]; ok && v.(*schema.Set).Len() > 0 {
+			vL := v.(*schema.Set).List()
+			overrides := make([]*ec2.LaunchTemplateOverrides, 0)
+
+			for _, v := range vL {
+				ors := v.(map[string]interface{})
+				lto := &ec2.LaunchTemplateOverrides{}
+
+				if v, ok := ors["availability_zone"].(string); ok && v != "" {
+					lto.AvailabilityZone = aws.String(v)
+				}
+
+				if v, ok := ors["instance_type"].(string); ok && v != "" {
+					lto.InstanceType = aws.String(v)
+				}
+
+				if v, ok := ors["spot_price"].(string); ok && v != "" {
+					lto.SpotPrice = aws.String(v)
+				}
+
+				if v, ok := ors["subnet_id"].(string); ok && v != "" {
+					lto.SubnetId = aws.String(v)
+				}
+
+				if v, ok := ors["weighted_capacity"].(float64); ok && v > 0 {
+					lto.WeightedCapacity = aws.Float64(v)
+				}
+
+				if v, ok := ors["priority"].(float64); ok {
+					lto.Priority = aws.Float64(v)
+				}
+
+				overrides = append(overrides, lto)
+			}
+
+			ltc.Overrides = overrides
+		}
+
+		configs = append(configs, ltc)
+	}
+
+	return configs, nil
+}
+
 func resourceAwsSpotFleetRequestCreate(d *schema.ResourceData, meta interface{}) error {
 	// http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotFleet.html
 	conn := meta.(*AWSClient).ec2conn
 
-	launch_specs, err := buildAwsSpotFleetLaunchSpecifications(d, meta)
-	if err != nil {
-		return err
+	_, launchSpecificationOk := d.GetOk("launch_specification")
+	_, launchTemplateConfigsOk := d.GetOk("launch_template_configs")
+
+	if !launchSpecificationOk && !launchTemplateConfigsOk {
+		return fmt.Errorf("One of `launch_specification` or `launch_template` must be set for a fleet request")
 	}
```

Comment:
Instead of this resource logic error handling during apply, we can now use the `Schema` `AtLeastOneOf` field to declaratively require one of these, potentially caught during plan-time. You should be able to combine both `AtLeastOneOf` and `ConflictsWith`/`ExactlyOneOf` 👍